BraveWallet/Crypto/Transaction Confirmations/TransactionConfirmationView.swift

```diff
@@ -239,16 +253,37 @@ struct TransactionConfirmationView: View {
             erc20ApproveHeader
           } else {
             TransactionHeader(
-              fromAccountAddress: confirmationStore.activeTransaction.fromAddress,
-              fromAccountName: fromAccountName,
-              toAccountAddress: confirmationStore.activeTransaction.ethTxToAddress,
-              toAccountName: toAccountName,
+              fromAccountAddress: confirmationStore.activeParsedTransaction?.fromAddress ?? "",
+              fromAccountName: confirmationStore.activeParsedTransaction?.namedFromAddress ?? "",
+              toAccountAddress: confirmationStore.activeParsedTransaction?.toAddress ?? "",
+              toAccountName: confirmationStore.activeParsedTransaction?.namedToAddress ?? "",
               originInfo: confirmationStore.state.originInfo,
               transactionType: transactionType,
               value: "\(confirmationStore.state.value) \(confirmationStore.state.symbol)",
               fiat: confirmationStore.state.fiat
             )
           }
+          
+          if confirmationStore.activeTransaction.txType == .solanaSplTokenTransferWithAssociatedTokenAccountCreation {
```

Comment:
I think is might be better kept in state similar to `isUnlimitedApprovalRequested`. 
`if confirmationStore.state.isSplTransferWithAssociatedTokenAccountCreation { ... }`